lib/icingadb/icingadb-itl.conf

```diff
@@ -0,0 +1,35 @@
+/* Icinga 2 | (c) 2022 Icinga GmbH | GPLv2+ */
+
+System.assert(Internal.run_with_activation_context(function() {
+	template CheckCommand "icingadb-check-command" use (checkFunc = Internal.IcingadbCheck) {
+		execute = checkFunc
+	}
+
+	object CheckCommand "icingadb" {
+		import "icingadb-check-command"
+
+		var criticalPendingSize = 1000000000
+		var objectSize = 1000
+		var maxObjectsPerQuery = 100
+		var additionalSmallerQueriesPerObject = 1
+		var criticalPendingQueries = (criticalPendingSize / (maxObjectsPerQuery * objectSize)) * (1 + additionalSmallerQueriesPerObject)
```

Comment:
Looks like you're estimating a number of queries to expected with a certain queue size in bytes. This makes me wonder: wouldn't be a good idea (possibly as a future extension) to also track the size of the queue in terms of used memory. I think that would make the number much more meaningful.